zone/raids.cpp

```diff
@@ -1543,21 +1543,23 @@ void Raid::SendRaidGroupRemove(const char *who, uint32 gid)
 
 void Raid::SendRaidMOTD(Client *c)
 {
-	if (!c || motd.empty()) {
+	if (!c || motd.empty() || c->IsBot()) {
 		return;
 	}
 
-	if (entity_list.GetBotByBotName(c->GetName())) {
-		return;
-	}
+	RaidMOTD_Struct temp;
+	temp.general.action = raidSetMotd;
+	temp.general.parameter = 0;
+	temp.general.unknown1 = 0;
+	strn0cpy(temp.general.leader_name, c->GetName(), 64);
+	strn0cpy(temp.general.player_name, GetLeaderName().c_str(), 64);
+	temp.motd = motd;
 
-	size_t size = motd.size() + 1;
-	auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(RaidMOTD_Struct) + size);
-	auto rmotd = (RaidMOTD_Struct *)outapp->pBuffer;
-	rmotd->general.action = raidSetMotd;
-	strn0cpy(rmotd->general.player_name, c->GetName(), 64);
-	strn0cpy(rmotd->motd, motd.c_str(), size);
-	c->FastQueuePacket(&outapp);
+	auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(RaidNote_Struct));
+	outapp->pBuffer = (unsigned char*)&temp;
+	QueuePacket(outapp);
+	outapp->pBuffer = nullptr;
+	safe_delete(outapp);
```

Comment:
This will leak the existing `outapp->pBuffer` (which is allocated to `sizeof(RaidNote_Struct)`) since it's being re-assigned.

Also doing it like this would be unsafe if this packet is ever queued since the `pBuffer` copied by `QueuePacket` would be left dangling once `temp` is gone. 

Since outapp's buffer should already be a `RaidMOTD_Struct` it can just be casted and modifed directly instead of using `temp`:

```cpp
RaidMOTD_Struct* temp = reinterpret_cast<RaidMOTD_Struct*>(outapp->pBuffer);
temp->general.action = raidSetMotd;
...
```

Reply:
This was kind of bad advice due to the struct having a `std::string` (ignoring that technically reinterpret_casting to the struct and then de-referencing it is already undefined behavior because it violates strict aliasing but we do that everywhere already).

Since the struct has the string it would have to actually be constructed to be used instead of just overlaying the memory like this. The struct would have to be created and then memcpy'd onto the buffer. But that's undefined behavior because it contains a std::string.